app/api/groups/get_group_progress.go

```diff
@@ -0,0 +1,175 @@
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+func (srv *Service) getGroupProgress(w http.ResponseWriter, r *http.Request) service.APIError {
+	user := srv.GetUser(r) // hack it with: user := database.NewUser(1, srv.Store.Users(), nil)
+
+	groupID, err := service.ResolveURLQueryPathInt64Field(r, "group_id")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	if apiError := checkThatUserOwnsTheGroup(srv.Store, user, groupID); apiError != service.NoError {
+		return apiError
+	}
+
+	itemParentIDs, err := service.ResolveURLQueryGetInt64SliceField(r, "parent_item_ids")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	userSelfGroupID, err := user.SelfGroupID()
+	if err == database.ErrUserNotFound {
+		return service.InsufficientAccessRightsError
+	}
+	service.MustNotBeError(err)
+
+	var found bool
+	found, err = srv.Store.Groups().OwnedBy(user).Where("groups.ID = ?", groupID).HasRows()
+	service.MustNotBeError(err)
+	if !found {
+		return service.InsufficientAccessRightsError
+	}
+
+	itemsVisibleToUserSubquery := srv.Store.GroupItems().
+		Select(
+			"idItem, MIN(sCachedFullAccessDate) <= NOW() AS fullAccess, "+
+				"MIN(sCachedPartialAccessDate) <= NOW() AS partialAccess, "+
+				"MIN(sCachedGrayedAccessDate) <= NOW() AS grayedAccess").
+		Joins(`
+			JOIN (SELECT * FROM groups_ancestors WHERE (groups_ancestors.idGroupChild = ?)) AS ancestors
+			ON ancestors.idGroupAncestor = groups_items.idGroup`, userSelfGroupID).
+		Group("groups_items.idItem").
+		Having("fullAccess > 0 OR partialAccess > 0 OR grayedAccess > 0").SubQuery()
+
+	// Preselect item IDs since we want to use them twice (for end members stats and for final stats)
+	// There should not be many of them
+	var itemIDs []int64
+	service.MustNotBeError(srv.Store.ItemItems().Where("idItemParent IN (?)", itemParentIDs).
+		Joins("JOIN ? AS visible ON visible.idItem = items_items.idItemChild", itemsVisibleToUserSubquery).
+		Pluck("items_items.idItemChild", &itemIDs).Error())
+
+	// Preselect IDs of groups for that we will calculate the final stats.
+	// All the "end members" are descendants of these groups.
+	// There should not be too many of groups because we paginate on them.
+	var ancestorGroupIDs []int64
+	ancestorGroupIDQuery := srv.Store.GroupGroups().
+		Where("groups_groups.idGroupParent = ?", groupID).
+		Joins(`
+			JOIN groups AS group_child
+			ON group_child.ID = groups_groups.idGroupChild AND group_child.sType NOT IN('Team', 'UserSelf')`)
+	ancestorGroupIDQuery, apiError := service.ApplySortingAndPaging(r, ancestorGroupIDQuery, map[string]*service.FieldSortingParams{
+		"id": {ColumnName: "group_child.ID", FieldType: "int64"},
+	}, "id")
+	if apiError != service.NoError {
+		return apiError
+	}
+	ancestorGroupIDQuery = service.SetQueryLimit(r, ancestorGroupIDQuery, 10, 20)
+	service.MustNotBeError(ancestorGroupIDQuery.
+		Pluck("group_child.ID", &ancestorGroupIDs).Error())
+
+	endMembersStats := srv.Store.
```

Comment:
This may be well simplified (and probably made more efficient (?)) with window function which are supported in MySQL since v8. It would save the inner join of `groups_attempts`. 
Do you confirm?

Reply:
It is something that we may want to check